mindsdb/integrations/handlers/gcs_handler/gcs_handler.py

```diff
@@ -0,0 +1,360 @@
+from contextlib import contextmanager
+
+import json
+import duckdb
+import pandas as pd
+import fsspec
+import google.auth
+from google.cloud import storage
+from typing import Text, Dict, Optional, List
+from duckdb import DuckDBPyConnection
+
+from mindsdb_sql.parser.ast.base import ASTNode
+from mindsdb_sql.parser.ast import Select, Identifier, Insert
+
+from mindsdb.utilities import log
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+
+from mindsdb.integrations.libs.api_handler import APIResource, APIHandler
+from mindsdb.integrations.utilities.sql_utils import FilterCondition
+
+logger = log.getLogger(__name__)
+
+
+class ListFilesTable(APIResource):
```

Comment:
Can we move the table classes to a different module called `gcs_tables.py`? That would make the code a little more modular.

Reply:
Done